### DISCUSSION TITLE
Writable computed inside a pinia store

### DISCUSSION THREAD

**Discussion by ajuvonen** [vuejs/pinia#2764] [Help and Questions] [answered]
How might one go about using writable computeds in a Pinia store with Vue3?

I'm able to create a computed that reads values and upon using it in conjunction with v-model also writes values to the correct place. Unfortunately, I haven't been able to create a testable solution for this, because storeToRefs doesn't return WritableComputed and there's apparently no alternate utility for it either.

My use case is, that I have a data structure with gaps, so my computed returns the data or an empty array, whereas in the set method it creates the needed structure if it was missing, otherwise just sets its value.

**Comment by posva** [comment]
Creating writable computed is fine, you can consume the store directly `store.myComputed` without using storeToRefs. You do need to return [writable state properties](https://pinia.vuejs.org/core-concepts/#Setup-Stores)

**Reply by ajuvonen** [reply]
Thank you for your help! This is actually interesting, because I now wrote a unit test directly assigning to the writable computed and it seems to work as expected - however I get a TS error "cannot assign to 'myComputed' because it is a read-only property", and the code won't build. Does this seem like a missing feature I could report somewhere?

**Reply by posva** [reply]
Can you [repro it here](https://play.pinia.vuejs.org/#eNqNU8Fu2zAM/RXBl6RoIrXI0AFBUmQbeugO27Du6ItjM6lSiRIk2TUQ+N9HS4nrdkPRm0Q+vvdIicfsi7W8qSFbZitfOmkD8xBqy1SB+3WeBZ9ntzlKbY0L7MhqDw/BOGAd2zmj2YSL0tQYwPHgJznmWBr0RBJB6wE/vchxJZIC8dElgLaqCEA3xlbbOgSDbFMqWT6RbqzneHlJ8vdYOtCApH9MxBxZ161EKiKClRixZbMs2Z3rwvKDN0jNHXuV/JSgnpYsRvqYlSiLPpJnIp45eD3fOvPsqasDgWdn6IYmJSpogjHKzwsrU9ljCNYvhSgrJHwFSjaOIwSBVot/ajY3/IZfCyW3goSExAra1zJUMa9Af4T9DN1c8etP/HNkbRYnUt2z9qRdjh2NJXh6nJ3cvxlKabSVCtxPGyQ93qvhFEqZ5+8xFlwNg8XyEcqn/8QPvk2ufzmg8TUwaisUbg8hpe8efkBL5yGpTVUrQr+T/A3eqLr3mGBfa6zI9ggX3d7HN5a4/+Pv2gDoz031RuM0Ij6O+ds7rb/YXfDFaIov/51GOCxGBTuJb3Yj/iZaigHkYDckST3uC7QxmdZmWK/1mHA6OWlOZmx6wda3yWQqQcIS7/SKNqyPOlpfhyRGO0KGKZp1fwEqtlBG) ? I will take a look

**Reply by ajuvonen** [reply]
[Here you go](https://play.pinia.vuejs.org/#eNqNVU1v2kAQ/SsjX+xIYDdKlUoUCEmbqu0hrZJeqroHYw+wwd5d7YehQvz3jnexMWkSFS727pv33szsjnfBtZRxbTEYBWOdKyYNaDRWQpnx5SQNjE6DacpZJYUysAOr8cEIhbCHhRIVhHGSC8sNqtjoMOUpzwXXROJAkw4fnaV8nHgF4qMXg5UsM4P0BjAuWD39KawiVsuMhnmm12hGsNuB54r9xo1bh/1+nDQhbSzUQ7Ygux46zzj9dVwiX5qVSwDo50Q+V9UAWJMlVhr+CAurrMZG57lYEiIrfgU+Ch4aKNkawaywujpaIO65NUZwmOUly9dPnVAhQqGoohiSm3ukxHNsiWHDSMhv63HiiXxmJzlO74QB5MIuV21oz8ILBvpVi6XVqyj0seEZObkuipaqJzxOer0JBoFv/rDKZPyoBaejsmsU08MGnRDqky9DGkjGWdaspEHinmPU1XCuxEbTGXkk8KCFzujcJQXWRohSDzPJfNjKGKlHSZIXnPAFlqxWMUeTcFkl/8TMLuPL+Dwp2TwhoYTxArenMhQxLLD6H/YWOnsTn7+N3znW+uJAWjWsDek+5Xsqi9F01Bds+aQouagkK1F9k4bRVTgpTlaWYvPVrRllsbOYrzBfP7P+qLfe9XeFVL4ae2mZTC3R+O3bhzvc0nO3WYnCloR+ZfMetSht49HDbiwvyHYP59x+cT1mfPlD324Nct0m1Rh11XB4V+YPr6R+tHsRX/SqeJweVMJuzBS4YPzJpHGniUZMB1K4GEBTbmuw6GDkw80h3DqYH0fd2Jr0qaPwoB4OIDqDydTb9SEn82bSiEW/2rszgDCTssTm4bh0uOG/z94fWY4DoDUaHUpC1RgdVE8uaZ2Vlu4tK8lXFLmtIwgmExol7Q0+dIoGNjFx3HxqICMaZIr69QIzOXlGju7261pwBa0AjPy+l9/7bBV9NBSnpvTJB136TbPpCxDs/wJFCy6E), I hope the repro works

**Reply by ajuvonen** [reply]
So I would be happy if one could directly assign to a writable computed exported by a Pinia store without TS errors, or if storeToRefs would type it as WritableComputedRef instead of ComputedRef, allowing you to set its .value. Preferably both. But if that's against good convention, I could just do the implementation without writable computeds.

**Reply by posva** [reply, marked as answer]
I opened https://github.com/vuejs/pinia/issues/2767 to track
